protocol/common/message_processor.go

```diff
@@ -479,6 +510,16 @@ func (p *MessageProcessor) sendDataSync(ctx context.Context, publicKey *ecdsa.Pu
 		return errors.Wrap(err, "failed to encrypt message")
 	}
 
+	// The shared secret needs to be handle before we send a message
+	// otherwise the topic might not be set up before we receive a message
+	if p.handleSharedSecrets != nil {
+		err := p.handleSharedSecrets([]*sharedsecret.Secret{messageSpec.SharedSecret})
+		if err != nil {
+			return err
+		}
+
+	}
+
```

Comment:
If the shared secret needs to be handled before we send a message, should an error be returned if the handler is not set?

Reply:
I take that in some cases you might not want to set up an handler, say fire & forget messages, in the case of scripts for example.